### DISCUSSION TITLE
Installation Instruction

### DISCUSSION THREAD

**Discussion by famfop** [sardemff7/purple-libnotify-plus#19] [Q&A]
I am trying to install your package but I'm not sure how to install the software. Can you maybe write a (very) short installation instruction? Thanks

**Comment by ghost** [comment]
I did:

1. git clone https://github.com/sardemff7/purple-libnotify-plus
2. cd purple-libnotify-plus
3. ./autogen.sh        # Not sure if this is necessary
4. ./configure
5. make
6. make install

If you get an error about libnotify missing at step 4, it means you need the libnotify development package (libnotify-dev in Ubuntu).  Just install that and rerun ./configure.

**Comment by plasmoduck** [comment]
Im trying to build purple-libnotify-plus and I keep getting this:

configure: error: Package requirements (purple-events >= 0.99.1) were not met:

Package 'purple-events', required by 'virtual:world', not found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

But I've already built purple-events and its installed

libtool: install: /usr/bin/install -c .libs/purple-events-exec.soT /usr/local/lib/purple-2/purple-events-exec.so
libtool: install: /usr/bin/install -c .libs/purple-events-exec.lai /usr/local/lib/purple-2/purple-events-exec.la
gmake --no-print-directory install-data-hook
cd /usr/local/lib/purple-2 && \
        rm purple-events.la purple-events-exec.la

OS is FreeBSD

**Comment by sardemff7** [comment]
You need to set `PKG_CONFIG_PATH=/usr/local/lib/pkgconfig` in environment since `/usr/local` is not searched by default in `pkg-config`
(Edited to correct the missing `lib`)

**Reply by plasmoduck** [reply]
Thanks, I got it compiled and enabled it in finch > plugins. On FreeBSD it's actually "export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig". When I get a message from my contacts, I get no notify-send notifications in dunst. Whats going on here?

**Reply by sardemff7** [reply]
I cannot say for sure. For Freedesktop.org notifications to work, you need all these conditions to be true:
- D-Bus is correctly configured for Finch and dunst to be on the same bus
- `purple-events` is correctly configured:
    - the global plugin settings have the wanted event enabled
    - the group the contact is in has the event enabled or unset
    - the contact has the event enabled or unset
- the correct conditions are met:
    - the related account is connected since more than 5 seconds
    - the contact is not blocked (or `purple-events` is set not to restrict blocked contacts)
    - the conversation is not focused (it may be harder to detect in Finch)

The focus detection would be my first suspect in your case, you may try to [drop the condition](https://github.com/sardemff7/purple-events/blob/master/src/plugin/utils.c#L45) since you are compiling by hand.
Otherwise, if this is a “real” error, messages should be visible in the debug window or something, or maybe on stderr (you may need to redirect).

**Reply by plasmoduck** [reply]
Thanks for your reply. How do I configure purple-events? Are you talking about Plugins > Events handling / Events to order?

**Reply by sardemff7** [reply]
Yes, the display name of the plugin is indeed “Events handling”. I don't remember how it's presented exactly in the UI (especially in Finch)
